### ISSUE TITLE
Adding breakpoint when paused on line exits PHP CLI with core dump

### ISSUE THREAD

**Issue by jonyo** [xdebug/vscode-php-debug#494]
When you run PHP on the command line, and it is paused on a breakpoint, any add or removal of breakpoints crashes PHP.  On the command line it says:
```Segmentation fault (core dumped)```

When running PHP from web it seems to be ok adding/removing breakpoints when it is paused.

PHP version:  7.2.34-9+ubuntu18.04.1+deb.sury.org+1
XDebug version:  v3.0.0
Adapter version: v1.14.7

Your launch.json:  

```
{
	"version": "0.2.0",
	"configurations": [
		{
			"name": "Listen for XDebug",
			"type": "php",
			"request": "launch",
			"port": 9003,
			"log":true,
			"pathMappings": {
				"/var/www/public/releases/local_source": "${workspaceRoot}"
			}
		}
	]
}
```

XDebug php.ini config:
```
zend_extension=xdebug.so
[xdebug]
xdebug.mode=develop,debug
xdebug.start_with_request=yes
xdebug.client_port=9005
xdebug.client_host=10.55.55.1
xdebug.log=/tmp/xdebug.log
```
XDebug logfile (from setting `xdebug.remote_log` in php.ini):  
```
[1299] Log opened at 2021-02-04 16:47:38.833659
[1299] [Step Debug] INFO: Connecting to configured address/port: 10.55.55.1:9003.
[1299] [Step Debug] INFO: Connected to debugging client: 10.55.55.1:9003 (through xdebug.client_host/xdebug.client_port). :-)
[1299] [Step Debug] -> <init xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" fileuri="file:///var/www/public/releases/local_source/app/webroot/test.php" language="PHP" xdebug:language_version="7.2.34-8+ubuntu18.04.1+deb.sury.org+1" protocol_version="1.0" appid="1299"><engine version="3.0.0"><![CDATA[Xdebug]]></engine><author><![CDATA[Derick Rethans]]></author><url><![CDATA[https://xdebug.org]]></url><copyright><![CDATA[Copyright (c) 2002-2020 by Derick Rethans]]></copyright></init>

[1299] [Step Debug] <- breakpoint_list -i 1
[1299] [Step Debug] -> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" command="breakpoint_list" transaction_id="1"></response>

[1299] [Step Debug] <- breakpoint_set -i 2 -t line -f file:///var/www/public/releases/local_source/app/webroot/test.php -n 3
[1299] [Step Debug] -> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" command="breakpoint_set" transaction_id="2" id="12990001"></response>

[1299] [Step Debug] <- breakpoint_list -i 3
[1299] [Step Debug] -> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" command="breakpoint_list" transaction_id="3"><breakpoint type="line" filename="file:///var/www/public/releases/local_source/app/webroot/test.php" lineno="3" state="enabled" hit_count="0" hit_value="0" id="12990001"></breakpoint></response>

[1299] [Step Debug] <- breakpoint_list -i 4
[1299] [Step Debug] -> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" command="breakpoint_list" transaction_id="4"><breakpoint type="line" filename="file:///var/www/public/releases/local_source/app/webroot/test.php" lineno="3" state="enabled" hit_count="0" hit_value="0" id="12990001"></breakpoint></response>

[1299] [Step Debug] <- run -i 5
[1299] [Step Debug] -> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" command="run" transaction_id="5" status="break" reason="ok"><xdebug:message filename="file:///var/www/public/releases/local_source/app/webroot/test.php" lineno="3"></xdebug:message></response>

[1299] [Step Debug] <- stack_get -i 6
[1299] [Step Debug] -> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" command="stack_get" transaction_id="6"><stack where="{main}" level="0" type="file" filename="file:///var/www/public/releases/local_source/app/webroot/test.php" lineno="3"></stack></response>

[1299] [Step Debug] <- context_names -i 7 -d 0
[1299] [Step Debug] -> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" command="context_names" transaction_id="7"><context name="Locals" id="0"></context><context name="Superglobals" id="1"></context><context name="User defined constants" id="2"></context></response>

[1299] [Step Debug] <- context_get -i 8 -d 0 -c 0
[1299] [Step Debug] -> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" command="context_get" transaction_id="8" context="0"></response>

[1299] [Step Debug] <- breakpoint_list -i 9
[1299] [Step Debug] -> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" command="breakpoint_list" transaction_id="9"><breakpoint type="line" filename="file:///var/www/public/releases/local_source/app/webroot/test.php" lineno="3" state="enabled" hit_count="1" hit_value="0" id="12990001"></breakpoint></response>

[1299] [Step Debug] <- breakpoint_remove -i 10 -d 12990001
```

Adapter logfile (from setting `"log": true` in launch.json):
```
-> setBreakpointsRequest
{
  command: 'setBreakpoints',
  arguments: {
    source: {
      name: 'test.php',
      path: '/Volumes/ZumbaEnv/public/app/webroot/test.php'
    },
    lines: [ 3, 4 ],
    breakpoints: [ { line: 3 }, { line: 4 } ],
    sourceModified: false
  },
  type: 'request',
  seq: 13
}

<- outputEvent
OutputEvent {
  seq: 0,
  type: 'event',
  event: 'output',
  body: { category: 'console', output: 'connection 1 closed\n' }
}

connection 1 closed
<- continuedEvent
ContinuedEvent {
  seq: 0,
  type: 'event',
  event: 'continued',
  body: { threadId: 1, allThreadsContinued: false }
}

<- threadEvent
ThreadEvent {
  seq: 0,
  type: 'event',
  event: 'thread',
  body: { reason: 'exited', threadId: 1 }
}
```

Code snippet to reproduce:

```php
<?php

echo 'on line 3';
echo 'on line 4';

```
To reproduce:  Add breakpoint on line 3.  Run it from command line like `php test.php`.  When it pauses on line 3, add a breakpoint on line 4.  PHP exits with seg fault.

**Comment by zobo** [comment]
I have a different version, PHP 8.0.0, Xdebug 3.0.1 and I can't reproduce the issue. If PHP crashes due to Xdebug a better place to report this would be https://bugs.xdebug.org/.
I suggest you try with latest Xdebug version first.
Best

**Comment by jonyo** [comment]
Thanks @zobo !

I thought of starting here because I have not changed PHP or XDebug versions recently but I saw the xdebug extension version was updated.

That said, you have a fair point, it is not normal for PHP to segfault, that means something at PHP / xdebug level is messing up.

Interestingly, sometimes this is also spit out when it exits:
```free(): double free detected in tcache 2```

Which looks like some kind of c++ shallow copy type error or something.  I'll try updating xdebug (version of PHP is locked down for parity with production at the moment and updating beyond patch release is no small task).
